bin/genome-env-next

```diff
@@ -0,0 +1,452 @@
+        find "$PWD" -iname '*sqlite3-dump' -or -iname '*sqlite3n-dump' | while read DB_TXT; do
+            regen_db "$DB_TXT"
+        done
+    fi
+
+    log.info ' done'
+}
+
+function setup_test_db {
+    local BASE_DIR="$1"
+    echo "=> Creating test database..."
+    eval "$($TESTDB database create --bash --owner $TESTDBSERVER_DB_USER --based-on $SNAPSHOT_NAME)"
+    if test -z "$TESTDBSERVER_DB_NAME"
+    then
+        fatal $BUILD_ERR "failed to create test database"
+    fi
+
+    if test -z "$DISABLE_MIGRATIONS"
+    then
+        echo "=> Migrating test database..."
+        (
+            cd "$BASE_DIR/sqitch/gms"
+            $SQITCH config core.pg.host     $TESTDBSERVER_DB_HOST
+            $SQITCH config core.pg.username $TESTDBSERVER_DB_USER
+            $SQITCH config core.pg.password $TESTDBSERVER_DB_PASS
+            $SQITCH config core.pg.db_name  $TESTDBSERVER_DB_NAME
+            if ! $SQITCH status | grep -q 'Nothing to deploy'
+            then
+                $SQITCH deploy
+            fi
+        )
+    fi
+}
+
+function setup_env {
+    log.info -e "\n=> Setting Up Test Environment..."
+
+    local BIN MODULE \
+        BASE_DIR="$1" \
+        SNAPSHOT_LIB="/gsc/scripts/opt/genome/current/user/lib/perl" \
+        SNAPSHOT_BIN="/gsc/scripts/opt/genome/current/user/bin"
+
+    # remove deployed bin directory from PATH
+    PATH="$(echo $PATH | tr : "\n" | $GREP -v "$SNAPSHOT_BIN" | tr '\n' : | sed 's/:$//')"
+    PATH=$UR_REPO/bin:$PATH
+    PATH=$WF_REPO/bin:$PATH
+    PATH=$BASE_DIR/bin:$PATH
+    export PATH
+
+    # remove deployed lib/perl directory from PATH
+    PERL5LIB="$(echo $PERL5LIB | tr : "\n" | $GREP -v "$SNAPSHOT_LIB" | tr '\n' : | sed 's/:$//')"
+
+    for MODULE in UR Workflow Genome
+    do
+        if genome-perl -M$MODULE -e 1 2> /dev/null
+        then
+            fatal $BUILD_ERR "still found $MODULE after removing deployed snapshot"
+        fi
+    done
+
+    PERL5LIB=$UR_REPO/lib:$PERL5LIB
+    PERL5LIB=$BASE_DIR/configuration-manager/lib:$PERL5LIB
+    PERL5LIB=$WF_REPO/lib:$PERL5LIB
+    PERL5LIB=$BASE_DIR/lib/perl:$PERL5LIB
+    export PERL5LIB
+
+    GENOME_CONFIG_PATH=$BASE_DIR/gms-confg/etc/gms
+
+    for MODULE in UR Workflow Genome
+    do
+        if ! genome-perl -M$MODULE -e 1 2> /dev/null
+        then
+            fatal $BUILD_ERR "failed to load $MODULE"
+        fi
+    done
+
+    for BIN in ur workflow genome
+    do
+        if which $BIN | $GREP -q "$SNAPSHOT_BIN"
+        then
+            fatal $BUILD_ERR "$BIN found in $SNAPSHOT_BIN"
+        fi
+    done
+
+    export GENOME_DS_GMSCHEMA_SERVER="dbname=${TESTDBSERVER_DB_NAME};host=${TESTDBSERVER_DB_HOST};port=${TESTDBSERVER_DB_PORT}"
+    export GENOME_DS_GMSCHEMA_LOGIN=$TESTDBSERVER_DB_USER
+    export GENOME_DS_GMSCHEMA_AUTH=$TESTDBSERVER_DB_PASS
+
+    hash -r
+}
+
+THIS="$(basename "$0")"
+
+GIT="$(type -p git)" || fatal $DEP_ERR "git is required"
+SQITCH="$(type -p sqitch)" || fatal $DEP_ERR "sqitch is required"
```

Comment:
Good point, I'll get it deployed; however, I think I'll do it via Puppet not `genome-snapshot-deps` since it's a development dependency only.